frame/instrumental-strategy-pablo/src/lib.rs

```diff
@@ -209,7 +285,57 @@ pub mod pallet {
 
 	impl<T: Config> Pallet<T> {
 		#[transactional]
-		fn do_rebalance(_vault_id: &T::VaultId) -> DispatchResult {
+		fn do_rebalance(vault_id: &T::VaultId) -> DispatchResult {
```

Comment:
I would separate the inner match arm's logic into their own functions, so you have something like:

```
match task {
	FundsAvailability::Withdrawable(balance) => {
	        Self::withdraw(...)
	} ...
```

Reply:
Got it! Will refactor